### PR TITLE
Make Newrelic a peer dependency

### LINKED ISSUE #31: Make newlic package a peer dependency
We shouldn't bother installing it if it's not being used. We should make the calling application handle the installation instead.

Things to do:

- [ ] Switch to a peer dependency
- [ ] If possible, gracefully fail with error if newrelic not installed
- [ ] Update docs on how to install newrelic as a peer dependency

### PR DESCRIPTION
**Pull request by elazzabi**
Fixes #31

### CODE REVIEW

File: src/newrelic/README.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # VIP Go New Relic integration
 
 ## Initialization
-In order to use the New Relic integration, you need to call it before starting your application (preferably the first line of your main `app.js` or `index.js`). New Relic needs to bootstrap itself before your application in order to work properly.
+New Relic's package (`newrelic`) is a peer dependency of this module. This means, in order to use it, you need to install it manually. This ensures we're not obliging you to use a specific version, but will use whatever version you're using already.
+
```
Comment (mjangda) [inline src/newrelic/README.md]: What if you don't have it installed? We should have instructions for that.

> If you don't have the New Relic package already installed, you can run the following:
>
> ```
> npm install --save newrelic@^4.8.0
> ```
> 
> ...

**Comment by mjangda** [comment]
In `newrelic.js`, we should wrap it's `require` in a try-catch to avoid fatals if the newrelic package isn't installed.

**Comment by elazzabi** [comment]
I was thinking about a way to check if the package exists before using the `require`, but it looks like the `try/catch` method is the easiest solution

**Comment by nickdaugherty** [comment]
Just a note, this has become a blocker for some public API stuff as we need to get logging working there before going much further.

File: __tests__/newrelic.tests.js [hunk 2]
```diff
@@ -32,8 +39,21 @@ describe( 'src/newrelic', () => {
 		} );
 	} );
 
-	describe( 'Should return expected values when all configuration is present', () => {
-		test( 'Should return newrelic module', () => {
+	describe( 'Environment variables are present', () => {
+		test( 'Should fail if `newrelic` module errors out', () => {
+			jest.doMock( 'newrelic', () => {
```
Comment (mjangda) [inline __tests__/newrelic.tests.js]: Same note about `doMock`

File: __tests__/newrelic.tests.js [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
+	const OLD_ENV_VARS = process.env;
 
-		// eslint-disable-next-line no-undef
-		afterEach( () => {
-			process.env = OLD_ENV_VARS;
-		} );
+	beforeEach( () => {
+		jest.resetModules();
+
+		jest.doMock( 'newrelic', () => ( {
```
Comment (mjangda) [inline __tests__/newrelic.tests.js]: We can just use `jest.mock` here; `doMock` is only needed if using `babel-jest`.

**Comment by nickdaugherty** [comment]
I think we should actually remove it from `peerDependencies`, as that causes issues of its own. More info here: https://github.com/Automattic/vip-go-node/issues/101

**Pull request merged**